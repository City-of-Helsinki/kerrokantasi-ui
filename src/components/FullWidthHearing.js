import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

import FormatRelativeTime from '../utils/FormatRelativeTime';
import Icon from '../utils/Icon';
import LabelList from './LabelList';
import Link from './LinkWithLang';
import MouseOnlyLink from './MouseOnlyLink';
import config from '../config';
import getAttr from '../utils/getAttr';
import { FormattedMessage } from 'react-intl';
import { getHearingURL, getHearingMainImageURL } from '../utils/hearing';

// eslint-disable-next-line import/no-unresolved
import defaultImage from '@city-images/default-image.svg';

const FullWidthHearing = ({ hearing, className = '', language, history, intl }) => {
  const {formatTime, formatDate} = intl;
  const backgroundImage = getHearingMainImageURL(hearing);
  const styles = {
    backgroundImage: backgroundImage ? `url(${backgroundImage})` : `url(${defaultImage})`,
  };
  const translationAvailable = !!getAttr(hearing.title, language, { exact: true });
  const availableInLanguageMessages = {
    fi: 'Kuuleminen saatavilla suomeksi',
    sv: 'Hörandet tillgängligt på svenska',
    en: 'Questionnaire available in English',
  };

  return (
    <div className={`fullwidth-hearing ${className}`}>
      <MouseOnlyLink
        className="fullwidth-hearing-image"
        style={styles}
        history={history}
        url={getHearingURL(hearing)}
        altText={getAttr(hearing.main_image.caption, language)}
      />
      <div className="fullwidth-hearing-header">
        <div className="fullwidth-hearing-title-wrap">
          <h3 className="h2 fullwidth-hearing-title">
            <Link to={{path: getHearingURL(hearing)}}>{getAttr(hearing.title, language)}</Link>
          </h3>
        </div>
        <div className="fullwidth-hearing-comments">
          <Icon name="comment-o" aria-hidden="true" />&nbsp;{hearing.n_comments}
          <span className="sr-only">
            {hearing.n_comments === 1 ? (
              <FormattedMessage id="hearingCardFullWidthComment" />
            ) : <FormattedMessage id="hearingCardFullWidthComments" />}
          </span>
        </div>
        <div className="fullwidth-hearing-times">
          <span>
            <FormatRelativeTime messagePrefix="timeOpen" timeVal={hearing.open_at} />.
          </span>{' '}
          <FormatRelativeTime
            messagePrefix="timeClose"
            timeVal={hearing.close_at}
            formatTime={formatTime}
            formatDate={formatDate}
            frontpage
          />
        </div>
        <div className="fullwidth-hearing-labels clearfix">
          <LabelList className="hearing-list-item-labellist" labels={hearing.labels} language={language} />
        </div>
        {!translationAvailable && (
          <div className="hearing-card-notice">
            <Icon name="exclamation-circle" aria-hidden="true" />
            <FormattedMessage id="hearingTranslationNotAvailable" />
            {config.languages.map(
              lang =>
                (getAttr(hearing.title, lang, { exact: true }) ? (
                  <div className="language-available-message" key={lang}>{availableInLanguageMessages[lang]}</div>
                ) : null)
            )}
          </div>
        )}
      </div>
    </div>
  );
};

FullWidthHearing.propTypes = {
  className: PropTypes.string,
  hearing: PropTypes.object,
  history: PropTypes.object,
  language: PropTypes.string,
  intl: PropTypes.object,
};

export default withRouter(FullWidthHearing);
