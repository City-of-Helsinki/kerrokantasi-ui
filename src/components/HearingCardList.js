import React from 'react';
import PropTypes from 'prop-types';
import {Row, Col} from 'react-bootstrap';
import HearingCard from './HearingCard';
import {hearingShape} from '../types';

const HearingCardList = ({hearings, language, intl}) =>
  <Row className="hearing-card-list">
    {hearings && hearings.map((hearing) =>
      <Col xs={12} md={4} key={hearing.id}>
        <HearingCard hearing={hearing} language={language} intl={intl}/>
      </Col>
    )}
  </Row>;

HearingCardList.propTypes = {
  hearings: PropTypes.arrayOf(hearingShape),
  language: PropTypes.string,
  intl: PropTypes.object,
};

export default HearingCardList;
