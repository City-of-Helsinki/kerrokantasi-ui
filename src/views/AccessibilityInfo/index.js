/* eslint-disable react/no-danger */
import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { Col, Row, Grid } from 'react-bootstrap';
import { injectIntl, intlShape } from 'react-intl';

import trackLink from '../../utils/trackLink';

function getContent(language) {
  if (typeof window === "undefined") return "";

  // Return correct file content depending on the chosen language. @city-i18n will handle correct files depending
  // on the selected themes.
  if (language === "fi") {
    // eslint-disable-next-line import/no-unresolved
    return require('@city-i18n/accessibility-info/accessibility.fi.md');
  }
  if (language === "sv") {
    // eslint-disable-next-line import/no-unresolved
    return require('@city-i18n/accessibility-info/accessibility.sv.md');
  }
  if (language === "en") {
    // eslint-disable-next-line import/no-unresolved
    return require('@city-i18n/accessibility-info/accessibility.en.md');
  }
  return 'Content not available in current language';
}

class AccessibilityInfo extends React.Component {
  componentDidMount() {
    trackLink();
  }

  render() {
    const pageContent = getContent(this.props.language);
    const { intl } = this.props;
    return (
      <Grid className="accessibility-page">
        <Helmet title={intl.formatMessage({ id: 'accessibilityPage' })} />
        <Row>
          <Col md={8}>
            <div dangerouslySetInnerHTML={{ __html: pageContent }} />
          </Col>
        </Row>
      </Grid>
    );
  }
}

AccessibilityInfo.propTypes = {
  language: PropTypes.string,
  intl: intlShape
};

export default injectIntl(connect((state) => ({ language: state.language }))(AccessibilityInfo));
