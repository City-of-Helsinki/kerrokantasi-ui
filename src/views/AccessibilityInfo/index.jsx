/* eslint-disable import/no-unresolved */
import { connect } from 'react-redux';
import { useIntl } from 'react-intl';
import React, { useEffect, useState } from 'react';
import AccessibilityFi from '@city-i18n/accessibility-info/accessibility.fi.md';
import AccessibilitySv from '@city-i18n/accessibility-info/accessibility.sv.md';
import AccessibilityEn from '@city-i18n/accessibility-info/accessibility.en.md';

import MarkdownPage from '../../components/MarkdownPage/MarkdownPage';

function getContent(language) {
  if (typeof window === 'undefined') return '';

  // Return correct file content depending on the chosen language. @city-i18n will handle correct files depending
  // on the selected themes.
  if (language === 'fi') {
    return AccessibilityFi;
  }
  if (language === 'sv') {
    return AccessibilitySv;
  }
  if (language === 'en') {
    return AccessibilityEn;
  }
  return 'Content not available in current language';
}

const AccessibilityInfo = () => {
  const intl = useIntl();
  const { locale } = intl;

  const pageContent = getContent(locale);

  const [markdown, setMarkdown] = useState();

  useEffect(() => {
    async function fetchData() {
      if (pageContent) {
        const response = await fetch(pageContent);
        const text = await response.text();

        setMarkdown(text);
      }
    }
    fetchData();
  }, [pageContent]);
  return (
    <div className='container accessibility-page'>
      <MarkdownPage title={intl.formatMessage({ id: 'accessibilityPage' })} markdown={markdown} />
    </div>
  );
};

export default connect()(AccessibilityInfo);
