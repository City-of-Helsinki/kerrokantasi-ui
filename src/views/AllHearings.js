import React from 'react';
import Helmet from 'react-helmet';
import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
import {connect} from 'react-redux';
import {fetchHearingList, fetchLabels} from '../actions';
import HearingList from '../components/HearingList';
import Col from 'react-bootstrap/lib/Col';
import Row from 'react-bootstrap/lib/Row';
import queryString from 'query-string';
import {get} from 'lodash';
import getAttr from '../utils/getAttr';

class AllHearings extends React.Component {
  /**
   * Return a promise that will, as it fulfills, have added requisite
   * data for the All Hearings view into the dispatch's associated
   * store.
   *
   * @param dispatch Redux Dispatch function
   * @return {Promise} Data fetching promise
   */
  constructor(props) {
    super(props);

    this.state = {
      sortBy: '-created_at',
      showOnlyOpen: false,
      isMobile: window.innerWidth < 992
    };
    this.handleResize = this.handleResize.bind(this);
  }

  static fetchData(dispatch, sortBy, searchTitle, labels) {
    const params = (
      searchTitle ?
        {title: searchTitle, ordering: sortBy, include: 'geojson'} :
        {ordering: sortBy, include: 'geojson'}
    );

    if (labels) {
      params.label = labels;
    }
    return dispatch(fetchHearingList("allHearings", "/v1/hearing/", params));
  }

  static fetchLabels(dispatch) {
    return dispatch(fetchLabels());
  }


  handleSort(newOrder) {
    const {dispatch} = this.props;
    this.setState({sortBy: newOrder});
    AllHearings.fetchData(dispatch, newOrder);
  }

  changeFilter(newFilter) {
    this.setState({hearingFilter: newFilter});
  }

  handleChangeFilter(filter) {
    this.changeFilter(filter);
  }

  handleSearch(event, searchTitle, labels) {
    if (event) {
      event.preventDefault();
    }
    const {dispatch} = this.props;
    const {sortBy} = this.state;
    const labelIds = labels ? labels.map((label) => label.id).toString() : null;
    AllHearings.fetchData(dispatch, sortBy, searchTitle, labelIds);
    AllHearings.updateQueryStringOnSearch(searchTitle, labels, labelIds);
  }

  static updateQueryStringOnSearch(searchTitle, labels, labelIds) {
    if (history.pushState) {
      const nextQuery = queryString.stringify({
        search: searchTitle || undefined,
        label: labels.map((label) => getAttr(label.label)) || undefined
      });
      const newSearch = searchTitle || labelIds ? `?${nextQuery}` : '';
      const newurl = `${location.protocol}//${window.location.host}${window.location.pathname}${newSearch}`;
      window.history.pushState({path: newurl}, '', newurl);
    }
  }

  toggleShowOnlyOpen() {
    if (this.state.showOnlyOpen) {
      this.setState({showOnlyOpen: false});
    } else {
      this.setState({showOnlyOpen: true});
    }
  }

  handleResize() {
    this.setState({ isMobile: window.innerWidth < 992 });
  }

  componentDidMount() {
    const {dispatch} = this.props;
    AllHearings.fetchLabels(dispatch);
    // Note: Fetch hearings after label ids are usable -> componentWillReceiveProps
    window.addEventListener('resize', this.handleResize);
  }

  componentWillReceiveProps(nextProps) {
    const shouldFetchHearings = !this.props.labels.length && nextProps.labels.length;

    if (shouldFetchHearings) {
      const {dispatch, language} = this.props;
      const {sortBy} = this.state;
      const queryLabels = [].concat(get(queryString.parse(location.search), 'label', []));
      const selectedLabels = nextProps.labels.filter(
        (label) => queryLabels.includes(getAttr(label.label, language))
      ).map((label) => label.id);

      AllHearings.fetchData(dispatch, sortBy, '', selectedLabels.toString());
    }
  }

  componentWillUnmount() {
    window.removeEventListener('resize', this.handleResize);
  }

  render() {
    const {formatMessage} = this.props.intl;
    const {isLoading, labels, language} = this.props;
    const {showOnlyOpen, isMobile} = this.state;
    const activeTab = this.props.params.tab ? this.props.params.tab : 'list';

    return (<div className="container">
      <Helmet title={formatMessage({id: 'allHearings'})}/>
      <h1 className="page-title"><FormattedMessage id="allHearings"/></h1>
      <Row>
        <Col md={8}>
          <HearingList
            hearings={this.props.hearings}
            isLoading={isLoading}
            labels={labels}
            handleChangeFilter={this.handleChangeFilter.bind(this)}
            handleSort={this.handleSort.bind(this)}
            handleSearch={this.handleSearch.bind(this)}
            language={language}
            activeTab={activeTab}
            showOnlyOpen={showOnlyOpen}
            toggleShowOnlyOpen={this.toggleShowOnlyOpen.bind(this)}
            isMobile={isMobile}
          />
        </Col>
      </Row>
    </div>);
  }
}

AllHearings.propTypes = {
  intl: intlShape.isRequired,
  dispatch: React.PropTypes.func,
  language: React.PropTypes.string, // To rerender when language changes
  hearings: React.PropTypes.object,
  isLoading: React.PropTypes.string,
  labels: React.PropTypes.object,
  params: React.PropTypes.object,
};

const mapStateToProps = (state) => ({
  hearings: state.hearingLists.allHearings.data,
  isLoading: state.hearingLists.allHearings.isFetching || state.labels.isFetching,
  labels: state.labels.data,
  language: state.language
});

const WrappedAllHearings = connect(mapStateToProps)(injectIntl(AllHearings));
// We need to re-hoist the static fetchData to the wrapped component due to react-intl:
WrappedAllHearings.fetchData = AllHearings.fetchData;
export default WrappedAllHearings;
