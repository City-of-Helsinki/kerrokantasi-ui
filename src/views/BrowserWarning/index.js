import React from 'react';
import { Helmet } from 'react-helmet';

function BrowserWarning() {
  return (
    <div className="container">
      <Helmet>
        <title>Kerrokantasi</title>
      </Helmet>
      <h1>Kerrokantasi</h1>
      <div className="row">
        <div className="col-xs-12">
          <div className="browser-warning-text-container">
            <p>
              Kerrokantasi-palvelu ei toimi Internet Explorer-selaimella.
            </p>
            <p>
              Käytä toista selainta, kuten
              <a href="https://www.google.com/intl/fi/chrome/" rel="noopener noreferrer" target="_blank">
                Chrome
              </a>,
              <a href="https://www.mozilla.org/fi/firefox/new/" rel="noopener noreferrer" target="_blank">
                Firefox
              </a> tai
              <a href="https://www.microsoft.com/fi-fi/edge" rel="noopener noreferrer" target="_blank">
                Edge
              </a>, ole hyvä.
            </p>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="col-xs-12">
          <div className="browser-warning-text-container">
            <p>
              Kerrokantasi (Säg din åsikt) tjänsten fungerar inte med Internet Explorer.
            </p>
            <p>
              Vänligen använd någon annan webbläsare t.ex.
              <a href="https://www.google.com/intl/sv/chrome/" rel="noopener noreferrer" target="_blank">
                Chrome
              </a>,
              <a href="https://www.mozilla.org/sv-SE/firefox/new/" rel="noopener noreferrer" target="_blank">
                Firefox
              </a> eller
              <a href="https://www.microsoft.com/sv-se/edge" rel="noopener noreferrer" target="_blank">
                Edge
              </a>.
            </p>
          </div>
        </div>
      </div>
      <div className="row">
        <div className="col-xs-12">
          <div className="browser-warning-text-container">
            <p>
              Kerrokantasi (Voice your opinion) service does not work with Internet Explorer.
            </p>
            <p>
              Please use another browser such as
              <a href="https://www.google.com/intl/en_us/chrome/" rel="noopener noreferrer" target="_blank">
                Chrome
              </a>,
              <a href="https://www.mozilla.org/en-US/firefox/new/" rel="noopener noreferrer" target="_blank">
                Firefox
              </a> or
              <a href="https://www.microsoft.com/en-us/edge" rel="noopener noreferrer" target="_blank">
                Edge
              </a>.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
}

export default BrowserWarning;
