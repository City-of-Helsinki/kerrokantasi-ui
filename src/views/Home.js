import React from 'react';
import PropTypes from 'prop-types';
import Helmet from 'react-helmet';
import {injectIntl, intlShape, FormattedMessage} from 'react-intl';
import {connect} from 'react-redux';
import {fetchHearingList} from '../actions';
import {getTopHearing, getOpenHearings} from '../selectors/hearing';
import {getUser} from '../selectors/user';
import {Col, Grid, Row} from 'react-bootstrap';
import FullWidthHearing from '../components/FullWidthHearing';
import HearingCardList from '../components/HearingCardList';
import orderBy from 'lodash/orderBy';
import OverviewMap from '../components/OverviewMap';
import Link from '../components/LinkWithLang';
import trackLink from '../utils/trackLink';
import CreateHearingButton from '../components/Hearings/CreateHearingButton';
import { isAdmin } from '../utils/user';
import config from '../config';
import {getFeedbackUrl} from '../utils/languageUtils';

export class Home extends React.Component {
  constructor(props) {
    super(props);

    this.state = {isMobile: typeof window !== 'undefined' && window.innerWidth < 768};
    this.handleResize = this.handleResize.bind(this);
  }
  /**
   * Return a promise that will, as it fulfills, have added requisite
   * data for the home view into the dispatch's associated store.
   *
   * @param dispatch Redux Dispatch function
   * @return {Promise} Data fetching promise
   */
  static fetchData(dispatch) {
    return Promise.all([
      dispatch(fetchHearingList('topHearing', '/v1/hearing', {ordering: '-n_comments', open: true, limit: 1})),
      dispatch(fetchHearingList('openHearings', '/v1/hearing', {open: true, include: 'geojson'})),
    ]);
  }

  componentDidMount() {
    Home.fetchData(this.props.dispatch);
    if (typeof window !== 'undefined') window.addEventListener('resize', this.handleResize);
    trackLink();
  }

  componentWillUnmount() {
    if (typeof window !== 'undefined') window.removeEventListener('resize', this.handleResize);
  }

  handleResize() {
    this.setState({isMobile: typeof window !== 'undefined' && window.innerWidth < 768});
  }

  render() {
    const {formatMessage} = this.props.intl;
    const {topHearing, openHearings, language, user, intl} = this.props;
    const {isMobile} = this.state;
    const hearingMap =
      openHearings && openHearings.data ? (
        <div className="map">
          <h2>
            <FormattedMessage id="open-hearings-on-map" />
          </h2>
          <OverviewMap
            enablePopups
            hearings={openHearings.data}
            style={{width: '100%', height: isMobile ? '70%' : 600}}
            mapElementLimit={10}
          />
        </div>
      ) : null;
    const heroStyle = {
      backgroundImage: "url('" + config.heroImageURL + "')",
    };

    return (
      <div>
        <section className="page-section page-section--welcome" style={heroStyle}>
          <Grid>
            <Row>
              <Col xs={10} md={8} className="welcome-content">
                <Helmet
                  title={formatMessage({id: 'welcome'})}
                  meta={[{name: "description", content: formatMessage({id: 'descriptionTag'})}]}
                />
                <h1>
                  <FormattedMessage id="welcome" />
                </h1>
                <p className="lead">
                  <FormattedMessage id="welcomeMessage" />
                </p>
              </Col>
            </Row>
          </Grid>
          <div className="welcome-koro__bottom" />
        </section>
        <section className="page-section page-section--hearing-card">
          <div className="container">
            <h2 className="page-title">
              <FormattedMessage id="openHearings" />
            </h2>
            {topHearing && <FullWidthHearing hearing={topHearing} language={language} intl={intl}/>}
            {openHearings && openHearings.data && topHearing &&
              !openHearings.isFetching && (
                <div className="list">
                  <HearingCardList
                    hearings={
                      orderBy(openHearings.data
                        .filter(hearing => hearing.id !== topHearing.id), ['close_at'], ['desc'])
                    }
                    language={language}
                    intl={intl}
                  />
                  <p className="text-center">
                    <Link to={{path: "/hearings/list"}} className="btn btn-default">
                      <FormattedMessage id="allHearings" />
                    </Link>
                  </p>
                </div>
              )}
          </div>
          <div className="hearings-koro__bottom" />
        </section>
        <section className="page-section page-section--feedback">
          <div className="container">
            <h2 className="feedback-prompt">
              <a href={getFeedbackUrl(language)} target="_blank" rel="noopener noreferrer" className="feedback-box">
                <FormattedMessage id="feedbackPrompt" />{' '}
              </a>
            </h2>
          </div>
        </section>
        <section className="page-section page-section--map">
          <div className="container">
            {hearingMap}
          </div>
        </section>
        {isAdmin(user) && <CreateHearingButton to={{path: '/hearing/new'}} />}
      </div>
    );
  }
}

Home.propTypes = {
  intl: intlShape.isRequired,
  dispatch: PropTypes.func,
  openHearings: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  topHearing: PropTypes.object, // eslint-disable-line react/forbid-prop-types
  language: PropTypes.string, // make sure changing language refreshes
  user: PropTypes.object,
};

const mapStateToProps = state => ({
  topHearing: getTopHearing(state),
  language: state.language,
  openHearings: getOpenHearings(state),
  user: getUser(state),
});

const WrappedHome = connect(mapStateToProps)(injectIntl(Home));
// We need to re-hoist the static fetchData to the wrapped component due to react-intl:
WrappedHome.fetchData = Home.fetchData;
export default WrappedHome;
