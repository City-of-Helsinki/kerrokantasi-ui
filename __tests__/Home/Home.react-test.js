import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {Home} from '../../src/views/Home';
import {getIntlAsProp, mockStore} from '../../test-utils';
import {MemoryRouter} from 'react-router-dom';
import HearingCardList from '../../src/components/HearingCardList';
import FullWidthHearing from '../../src/components/FullWidthHearing';

// Renders the Hearings component using enzymes shallow rendering.
// You can pass props you want to override as a parameter.
const setup = propOverrides => {
  const {hearingLists: {allHearings}, ...rest} = mockStore;
  const props = Object.assign({
    topHearing: allHearings.data[0],
    openHearings: allHearings,
    ...rest
  }, propOverrides);

  const wrapper = shallow(<MemoryRouter><Home intl={getIntlAsProp()} {...props} /></MemoryRouter>);

  return {
    props,
    wrapper
  };
};

test('Home component should render as expected', () => {
  const {wrapper} = setup();
  const tree = toJson(wrapper.dive().dive());
  expect(tree).toMatchSnapshot();
});

test('Should find top hearing', () => {
  const {wrapper} = setup();
  expect(wrapper.dive().dive().find(FullWidthHearing)).toHaveLength(1);
});

test('Should find HearingCardList', () => {
  const {wrapper} = setup();
  expect(wrapper.dive().dive().find(HearingCardList)).toHaveLength(1);
});
