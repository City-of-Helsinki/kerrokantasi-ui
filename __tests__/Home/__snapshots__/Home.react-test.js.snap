// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component should render as expected 1`] = `
<div>
  <section
    className="page-section page-section--welcome"
    style={
      Object {
        "backgroundImage": "url('http://materialbank.myhelsinki.fi/detail/1192/download/7')",
      }
    }
  >
    <Grid
      bsClass="container"
      componentClass="div"
      fluid={false}
    >
      <Row
        bsClass="row"
        componentClass="div"
      >
        <Col
          bsClass="col"
          className="welcome-content"
          componentClass="div"
          md={8}
          xs={10}
        >
          <HelmetWrapper
            defer={true}
            encodeSpecialCharacters={true}
            title="Kerro mielipiteesi"
          />
          <h1>
            <FormattedMessage
              id="welcome"
              values={Object {}}
            />
          </h1>
          <p
            className="lead"
          >
            <FormattedMessage
              id="welcomeMessage"
              values={Object {}}
            />
          </p>
        </Col>
      </Row>
    </Grid>
    <div
      className="welcome-koro__bottom"
    />
  </section>
  <section
    className="page-section page-section--hearing-card"
  >
    <div
      className="container"
    >
      <h2
        className="page-title"
      >
        <FormattedMessage
          id="openHearings"
          values={Object {}}
        />
      </h2>
      <withRouter(FullWidthHearing)
        hearing={
          Object {
            "abstract": Object {
              "fi": "Tämä on suomenkielinen testiabstrakti. Sisältää huikean kuvauksen kyseisestä kuulemisesta.",
            },
            "borough": Object {},
            "close_at": "2200-11-28T22:00:00Z",
            "closed": false,
            "created_at": "2017-10-23T11:07:26.630423Z",
            "default_to_fullscreen": false,
            "geojson": null,
            "id": "rXT2L1HxEOZTERjluyxyQZ412aYM8oZE",
            "labels": Array [],
            "main_image": Object {
              "caption": Object {
                "fi": "A Wonderfull caption",
              },
              "height": 631,
              "id": 92,
              "title": Object {},
              "url": "https://api.hel.fi/kerrokantasi-test/media/images/2017/10/LzqVMrMI.jpeg",
              "width": 893,
            },
            "n_comments": 1,
            "open_at": "2017-10-22T21:00:00Z",
            "organization": "Mock Inc",
            "published": true,
            "servicemap_url": "",
            "slug": "mockHearing",
            "title": Object {
              "fi": "Mock Hearing",
            },
          }
        }
      />
      <div
        className="list"
      >
        <HearingCardList
          hearings={Array []}
        />
        <p
          className="text-center"
        >
          <withRouter(Connect(LinkWithLangComponent))
            className="btn btn-default"
            to={
              Object {
                "path": "/hearings/list",
              }
            }
          >
            <FormattedMessage
              id="allHearings"
              values={Object {}}
            />
          </withRouter(Connect(LinkWithLangComponent))>
        </p>
      </div>
    </div>
    <div
      className="hearings-koro__bottom"
    />
  </section>
  <section
    className="page-section page-section--feedback"
  >
    <div
      className="container"
    >
      <a
        className="feedback-box"
        href="mailto:kerrokantasi@example.com?subject=Kerrokantasi-palaute"
      >
        <h2
          className="feedback-prompt"
        >
          <FormattedMessage
            id="feedbackPrompt"
            values={Object {}}
          />
        </h2>
      </a>
    </div>
  </section>
  <section
    className="page-section page-section--map"
  >
    <div
      className="container"
    >
      <div
        className="map"
      >
        <h2>
          <FormattedMessage
            id="open-hearings-on-map"
            values={Object {}}
          />
        </h2>
        <Connect(OverviewMap)
          enablePopups={true}
          hearings={
            Array [
              Object {
                "abstract": Object {
                  "fi": "Tämä on suomenkielinen testiabstrakti. Sisältää huikean kuvauksen kyseisestä kuulemisesta.",
                },
                "borough": Object {},
                "close_at": "2200-11-28T22:00:00Z",
                "closed": false,
                "created_at": "2017-10-23T11:07:26.630423Z",
                "default_to_fullscreen": false,
                "geojson": null,
                "id": "rXT2L1HxEOZTERjluyxyQZ412aYM8oZE",
                "labels": Array [],
                "main_image": Object {
                  "caption": Object {
                    "fi": "A Wonderfull caption",
                  },
                  "height": 631,
                  "id": 92,
                  "title": Object {},
                  "url": "https://api.hel.fi/kerrokantasi-test/media/images/2017/10/LzqVMrMI.jpeg",
                  "width": 893,
                },
                "n_comments": 1,
                "open_at": "2017-10-22T21:00:00Z",
                "organization": "Mock Inc",
                "published": true,
                "servicemap_url": "",
                "slug": "mockHearing",
                "title": Object {
                  "fi": "Mock Hearing",
                },
              },
            ]
          }
          style={
            Object {
              "height": 600,
              "width": "100%",
            }
          }
        />
      </div>
    </div>
  </section>
</div>
`;
